metron-analytics/metron-profiler-client/src/main/java/org/apache/metron/profiler/client/stellar/GetProfile.java

```diff
@@ -0,0 +1,179 @@
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+import static org.apache.metron.common.dsl.Context.Capabilities.PROFILER_COLUMN_BUILDER;
+import static org.apache.metron.common.dsl.Context.Capabilities.PROFILER_HBASE_TABLE;
+import static org.apache.metron.common.dsl.Context.Capabilities.PROFILER_ROW_KEY_BUILDER;
+
+/**
+ * A Stellar function that can retrieve data contained within a Profile.
+ *
+ *  PROFILE_GET
+ *
+ * Retrieve all values for 'entity1' from 'profile1' over the past 4 hours.
+ *
+ *   <code>PROFILE_GET('profile1', 'entity1', 4, 'HOURS')</code>
+ *
+ * Retrieve all values for 'entity1' from 'profile1' over the past 2 days.
+ *
+ *   <code>PROFILE_GET('profile1', 'entity1', 2, 'DAYS')</code>
+ *
+ * Retrieve all values for 'entity1' from 'profile1' that occurred on 'weekdays' over the past month.
+ *
+ *   <code>PROFILE_GET('profile1', 'entity1', 1, 'MONTHS', 'weekdays')</code>
+ *
+ */
+@Stellar(
+        namespace="PROFILE",
+        name="GET",
+        description="Retrieves a series of values from a stored profile.",
+        params={
+          "profile - The name of the profile.",
+          "entity - The name of the entity.",
+          "durationAgo - How long ago should values be retrieved from?",
+          "units - The units of 'durationAgo'.",
+          "groups - Optional - The groups used to sort the profile."
+        },
+        returns="The profile measurements."
+)
+public class GetProfile implements StellarFunction {
+
+  private ProfilerClient client;
+
+  /**
+   * Initialization.
+   */
+  @Override
+  public void initialize(Context context) {
+
+    Context.Capabilities[] required = { PROFILER_HBASE_TABLE, PROFILER_ROW_KEY_BUILDER, PROFILER_COLUMN_BUILDER };
+    validateContext(context, required);
+
+    HTableInterface table = (HTableInterface) context.getCapability(PROFILER_HBASE_TABLE).get();
+    RowKeyBuilder rowKeyBuilder = (RowKeyBuilder) context.getCapability(PROFILER_ROW_KEY_BUILDER).get();
+    ColumnBuilder columnBuilder = (ColumnBuilder) context.getCapability(PROFILER_COLUMN_BUILDER).get();
+
+    client = new HBaseProfilerClient(table, rowKeyBuilder, columnBuilder);
+  }
+
+  /**
+   * Is the function initialized?
+   */
+  @Override
+  public boolean isInitialized() {
+    return client != null;
+  }
+
+  /**
+   * Apply the function.
+   * @param args The function arguments.
+   * @param context
+   */
+  @Override
+  public Object apply(List<Object> args, Context context) throws ParseException {
+
+    String profile = getArg(0, String.class, args);
+    String entity = getArg(1, String.class, args);
+    long durationAgo = getArg(2, Long.class, args);
+    String unitsName = getArg(3, String.class, args);
+    TimeUnit units = TimeUnit.valueOf(unitsName);
+    List<Object> groups = getGroupsArg(4, args);
+
+    return client.fetch(profile, entity, durationAgo, units, Integer.class, groups);
```

Comment:
Yes, I think that would be a good enhancement.  But right now it assumes Numeric.  Mainly because of serialization.  Opened METRON-415 to track the enhancement of supporting a variety of types as profile values.